src/software/world/robot.cpp

```diff
@@ -25,6 +25,32 @@ Robot::Robot(RobotId id, const RobotState &initial_state, const Timestamp &times
 {
 }
 
+Robot::Robot(const TbotsProto::Robot &robot_proto)
+    : id_(robot_proto.id()),
+      current_state_(RobotState(robot_proto.current_state())),
+      timestamp_(Timestamp::fromTimestampProto(robot_proto.timestamp()))
+{
+    for (int i = 0; i < robot_proto.unavailable_capabilities_size(); i++)
+    {
+        auto unavailable_capability = robot_proto.unavailable_capabilities(i);
```

Comment:
we should be able to do a for each loop on this, e.g. `for (const auto & unavailable_capability : robot_proto.unavailable_capabilities())`

Reply:
Updated to foreach loop.